### PR TITLE
ci: switch to pure mocha & get coverage

### LINKED ISSUE #4: Remap?
Looks like current voidTrader is remapped when it's requested.... i think we should either:
- request a remap to this format on worldstate-parser
- keep the same original data mapping as the api

current remapping:
```js
{
  id: trader.id,
  from: new Date(trader.activation),
  until: new Date(trader.expiry),
  name: trader.character,
  relay: trader.location,
  goodies: trader.inventory,
  active: trader.active,
  fromString: `${name} arrives in ${trader.startString}`,
  untilString: `${name} leaves in ${trader.endString}`,
}
```

### LINKED ISSUE #5: Request-promise depreceated
[Request promise](https://www.npmjs.com/package/request-promise) is deprecated.
Is there any specific need to use it that i'm missing? Or would it be possible to change to [axios](https://github.com/axios/axios) or any other non deprecated library?

### LINKED ISSUE #8: Test page doesn't work
looks like the cdn location doesn't work anymore, and I don't know how that particular one is set up

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? (provide a description or issue closes statement)
continue work getting coverage
closes #4 
closes #5 
closes #8 

---

### Reproduction steps
[Actions](https://github.com/WFCD/Warframe.js/actions/workflows/ci.yaml)

---

### Considerations
- Does this contain a new dependency? **[Yes]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Enhancement]**

### CODE REVIEW

**Comment by TobiTenno** [comment]
I still need to:
- [ ] .min.js file for browser usage/compatibility
- [ ] pull in types better

**Comment by TobiTenno** [comment]
currently looking at using uglify.js for minification at publish time and writing back the tagged commit with the edited app.min.js

currently trying to sort out how to remove the require line so it's not included in the minification, since node-fetch is just the node-side implementation of window.Fetch

**Review by lucakiebel — DISMISSED** [review]
I really like the streamlined get function. I wanted TypeScript mostly for typehinting when integrating into other node projects, though we can work around this of course

**Comment by lucakiebel** [comment]
> currently looking at using uglify.js for minification at publish time and writing back the tagged commit with the edited app.min.js
> 
> currently trying to sort out how to remove the require line so it's not included in the minification, since node-fetch is just the node-side implementation of window.Fetch

@AyAyEm suggested [axios/axios](https://github.com/axios/axios), It also has a browser version, so could be either included in minification or just be added via script tag in the HTML, though I don't like the overhead. It's 5.3 KB minified

**Comment by TobiTenno** [comment]
That's still pretty minimal. The rain I like node-fetch is that it's a direct implementation of the existing browser API, so no overhead

**Comment by lucakiebel** [comment]
If we can make fetch work that is of course preferential.

**Comment by TobiTenno** [comment]
I got the line removals working, should only remove unneeded lines before minification, and I got the minification pre-publish working

File: warframe.js [hunk 1]
```diff
@@ -1,323 +1,138 @@
...
- * Warframe.js by Luca Kiebel
- *
- * Docs and more: https://github.com/lucakiebel/Warframe.js
- */
+'use strict';
 
-const request = require('request-promise');
+const fetch = require('node-fetch');
+
+const BASE = 'https://api.warframestat.us';
```
Comment (TobiTenno) [inline warframe.js]: my only worry for web usage is this :point_up: I don't really like making extraneous globals, but i couldn't find a good way without making more class stuff... i might try it as a pseudo-static var so we don't expose excessive amounts of vars

Comment (TobiTenno) [reply warframe.js]: I moved it in, but I'd love feedback. I wanted to make it a static const, but that'd require doing babel and stuff

**Review by lucakiebel — DISMISSED** [review]
Really like this, especially the minification, looks great! 👍🏻

**Comment by TobiTenno** [comment]
sorry, had one more cleanup, if you wouldn't mind reapproving :cry:

**Review by lucakiebel — APPROVED** [review]

**Pull request merged**